x-pack/plugins/ingest_manager/server/services/agents/upgrade.ts

```diff
@@ -80,7 +83,10 @@ export async function sendUpgradeAgentsActions(
           })
         ).agents;
   const agentsToUpdate = agents.filter(
-    (agent) => !agent.unenrollment_started_at && !agent.unenrolled_at
+    (agent) =>
+      !agent.unenrollment_started_at &&
+      !agent.unenrolled_at &&
+      isAgentUpgradeable(agent, kibanaVersion)
```

Comment:
Do you think we can update `isAgentUpgradeable` to have this logic?

Reply:
Good idea.  For the single agent upgrade handler I will still check it to return a descriptive error message.